test/sparse/sparse.jl

```diff
@@ -1853,120 +1853,49 @@ end
-    @test isequal(cov_sparse[1, 1:end], cov_dense[1, 1:end])
-
-    cov_sparse = cov(x_sparse, 2, corrected=true)
-    cov_sparse[isinf.(cov_sparse)] = NaN
-    cov_dense = cov(x_dense, 2, corrected=true)
-    @test cov_sparse[2:end, 2:end] ≈ cov_dense[2:end, 2:end]
-    @test isequal(cov_sparse[1:end, 1], cov_dense[1:end, 1])
-    @test isequal(cov_sparse[1, 1:end], cov_dense[1, 1:end])
-
-    cov_sparse = cov(x_sparse, 1, corrected=false)
-    cov_sparse[isinf.(cov_sparse)] = NaN
-    cov_dense = cov(x_dense, 1, corrected=false)
-    @test cov_sparse[2:end, 2:end] ≈ cov_dense[2:end, 2:end]
-    @test isequal(cov_sparse[1:end, 1], cov_dense[1:end, 1])
-    @test isequal(cov_sparse[1, 1:end], cov_dense[1, 1:end])
-
-    cov_sparse = cov(x_sparse, 2, corrected=false)
-    cov_sparse[isinf.(cov_sparse)] = NaN
-    cov_dense = cov(x_dense, 2, corrected=false)
-    @test cov_sparse[2:end, 2:end] ≈ cov_dense[2:end, 2:end]
-    @test isequal(cov_sparse[1:end, 1], cov_dense[1:end, 1])
-    @test isequal(cov_sparse[1, 1:end], cov_dense[1, 1:end])
-
-    # Test with NaN and Inf
-    x_sparse[2,1] = NaN
-    x_dense[2,1]  = NaN
-
-    cov_sparse = cov(x_sparse, 1, corrected=true)
-    cov_dense = cov(x_dense, 1, corrected=true)
-    @test cov_sparse[2:end, 2:end] ≈ cov_dense[2:end, 2:end]
-    @test isequal(cov_sparse[1:end, 1], cov_dense[1:end, 1])
-    @test isequal(cov_sparse[1, 1:end], cov_dense[1, 1:end])
-
-    cov_sparse = cov(x_sparse, 2, corrected=true)
-    cov_sparse[isinf.(cov_sparse)] = NaN
-    cov_dense = cov(x_dense, 2, corrected=true)
-    @test cov_sparse[3:end, 3:end] ≈ cov_dense[3:end, 3:end]
-    @test isequal(cov_sparse[1:end, 1:2], cov_dense[1:end, 1:2])
-    @test isequal(cov_sparse[1:2, 1:end], cov_dense[1:2, 1:end])
-
-    cov_sparse = cov(x_sparse, 1, corrected=false)
-    cov_dense = cov(x_dense, 1, corrected=false)
-    @test cov_sparse[2:end, 2:end] ≈ cov_dense[2:end, 2:end]
-    @test isequal(cov_sparse[1:end, 1], cov_dense[1:end, 1])
-    @test isequal(cov_sparse[1, 1:end], cov_dense[1, 1:end])
-
-    cov_sparse = cov(x_sparse, 2, corrected=false)
-    cov_sparse[isinf.(cov_sparse)] = NaN
-    cov_dense = cov(x_dense, 2, corrected=false)
-    @test cov_sparse[3:end, 3:end] ≈ cov_dense[3:end, 3:end]
-    @test isequal(cov_sparse[1:end, 1:2], cov_dense[1:end, 1:2])
-    @test isequal(cov_sparse[1:2, 1:end], cov_dense[1:2, 1:end])
+    if elty <: Real
+        nzvals = randn(np2)
+    else
+        nzvals = complex.(randn(np2), randn(np2))
+    end
+    x_sparse = sparse(rand(1:n, np2), rand(1:p, np2), nzvals, n, p)
+    x_dense  = convert(Matrix{elty}, x_sparse)
+    @testset "Test with no Infs and NaNs, vardim=$vardim, corrected=$corrected" for vardim in (1, 2),
+                                                                                 corrected in (true, false)
+        @test cov(x_sparse, vardim, corrected=corrected) ≈
+                cov(x_dense, vardim, corrected=corrected)
```

Comment:
I would say either 4 spaces from the previous level, or aligned with the `cov` above. Not sure which convention is more common.